File: java/dubbo-demo/dubbo-demo-api/src/main/java/com/alibaba/dubbo/demo/TypeRequest.java
package com.alibaba.dubbo.demo;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Map;

public class TypeRequest implements Serializable {
    private Map<String, String> map;

    private BigDecimal bigDecimal;

    public Map<String, String> getMap() {
        return map;
    }

    public void setMap(Map<String, String> map) {
        this.map = map;
    }

    public BigDecimal getBigDecimal() {
        return bigDecimal;
    }

    public void setBigDecimal(BigDecimal bigDecimal) {
        this.bigDecimal = bigDecimal;
    }
}

File: packages/dubbo/src/__tests__/providers/com/alibaba/dubbo/demo/TypeRequest.ts
import java from 'js-to-java';

//generate by interpret-cli dubbo2.js

export interface ITypeRequest {
  bigDecimal?: {value: string};
  map?: {[name: string]: string};
}

export class TypeRequest {
  bigDecimal?: {value: string};
  map?: {[name: string]: string};

  constructor(params: ITypeRequest) {
    this.bigDecimal = params.bigDecimal;
    this.map = params.map;
  }

  __fields2java() {
    let mapMapTransfer = new Map();
    for (let mapKey in this.map) {
      mapMapTransfer.set(java.String(mapKey), java.String(this.map[mapKey]));
    }
    return {
      $class: 'com.alibaba.dubbo.demo.TypeRequest',
      $: {
        bigDecimal: this.bigDecimal
          ? java.BigDecimal(this.bigDecimal.value)
          : null,
        map: java.Map(mapMapTransfer),
      },
    };
  }
}
